enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,725 @@
+
+#### Update-service-side filtering
+
+Instead of filtering cluster-side in the cluster-version operator, we could filter edges on the update-service side by querying [uploaded Telemetry][uploaded-telemetry] or [client-provided query parameters][cincinnati-for-openshift-request].
+However, there is more data available in-cluster beyond what is uploaded as Telemetry.
+And because we are [supporting edges which we do not recommend][openshift-docs-32091], we'd need to pass the reasons for not recommending those edges out to clusters anyway.
+Passing enough information to make the decision completely on the cluster side is not that much more work.
+
+Also in this space is clusters which are on restricted networks.
+Those clusters could be reconfigured to ship their Telemetry or Insights uploads to local aggregators, or could have their Telemetry and Insights sneakernetted to Red Hat's central aggregators.
+But client side filtering will work in restricted-network clusters without the need for any of that, especially now that [the OpenShift Update Service][osus] is making it easier to get Cincinnati responses to clusters in restricted networks.
+
+### Failing open
+
+Whether a conditional edge should be recommended for a given cluster depends on intent flowin from the graph-data maintainers, through update services, to the cluster version operator, and then being evaluated in-cluster.
+That flow can break down at any point; for example, the update service may only understand graph-data schema 1.0.0, and not understand 1.1.0.
+Or the cluster-version operator may have trouble connecting to the in-cluster Thanos/Prometheues.
+In those situations, this enhancement proposal recommends blocking the edge.
+
+An alternative approach would have been failing open, where "failed to evaluate the graph-data maintainer intentions" would result in recommending edges.
+That would reduce the risk of leaving a cluster stuck without any recommended updates.
+But evalution failures should trigger alerts, so the appropriate administrators can resolve the issue, and delaying an update until we can make a clear determination is safer than updating while we are unable to make a clear determination.
+
+As a final safety valve for situations where recovering evaluation capability would take too long, confident cluster administrators can force through the update guard.
+
+### Query coverage
+
+[The `promql` proposal](#enhanced-graph-data-schema-for-blocking-edges) specifies a single query that allows the update service to distinguish clusters where the edge is recommended (the query returns 0) from clusters where the edge is not recommended (the query returns 1).
+This allows the cluster-version operator to distinquish between the three states of `Recommended=True` (0), `Recommended=False` (1), or `Recommended=Unknonwn` (no result, for example because the query asked for metrics which the local Prometheus was failing to scrape).
```

Comment:
not clear to me how this is an alternative, it sounds like it's just describing the proposal

Reply:
Yes, I think I would move this paragraph up to the risks section since it explains the mitigation for any query failure.

Reply:
In one of the earlier PRs in this effort, there was some pushback about why the queries had to be complicated enough to return 1 or 0.  This alternatives section is my attempt to explain why that complication is worth the trouble.  I've added a comparison to a simpler `promql: cluster_infrastructure_provider{type=~"VSphere|None"}` alternative to this subsection with 6f093a0 -> 02d2825, and kept it in the alternatives section.  The risk section already includes a *Clusters without local Prometheus stacks* subsection, and I've added a paragraph there linking over to the alternatives subsection as related content.